src/authz-module/libraries-manager/context.tsx

```diff
@@ -0,0 +1,65 @@
+import React, {
+  createContext, useContext, useMemo, ReactNode,
+} from 'react';
+import { useParams } from 'react-router-dom';
+import { AppContext } from '@edx/frontend-platform/react';
+import { useLibrary, useValidateTeamMember } from './data/hooks';
+
+export type AppContextType = {
+  authenticatedUser: {
+    username: string;
+    email: string;
+  };
+};
+
+type LibraryAuthZContextType = {
+  canManageTeam: boolean;
+  username: string;
+  libraryId: string;
+  roles: string[];
+  permissions: string[];
+  libraryName: string;
+  libraryOrg: string;
+};
+
+const LibraryAuthZContext = createContext<LibraryAuthZContextType | undefined>(undefined);
+
+type AuthZProviderProps = {
+  children: ReactNode;
+};
+
+export const LibraryAuthZProvider: React.FC<AuthZProviderProps> = ({ children }) => {
+  const { libraryId } = useParams<{ libraryId: string }>();
+  const { authenticatedUser } = useContext(AppContext) as AppContextType;
+
+  if (!libraryId) {
+    throw new Error('MissingLibrary');
+  }
+  const { data: currentUser } = useValidateTeamMember(libraryId, authenticatedUser.username);
+
+  const { data: libraryMetadata } = useLibrary(libraryId);
+
+  const value = useMemo((): LibraryAuthZContextType => ({
+    username: currentUser.username,
+    libraryId,
+    libraryName: libraryMetadata.title,
+    libraryOrg: libraryMetadata.org,
+    roles: [],
+    permissions: [],
+    canManageTeam: currentUser.roles.includes('admin'),
+  }), [libraryId, authenticatedUser]);
```

Comment:
I've noticed that you’re already using react query for some hooks Im just wondering, in the provider, should we add libraryMetadata and currentUser as deps to the useMemo? I mean right now it only lists libraryId and authenticatedUser, so I was curious if that could ever cause stale values, also, have you thought about giving validateTeamMember/useLibrary a small staleTime like you did for useTeamMembers?

Reply:
Also, dependencies are compared using `Object.is` which doesn't do deep equality checking, so it's better to break things down to simple values, i.e. use `currentUser.username` (a simple `string` value) as the dependency here, not `authenticatedUser`.

```suggestion
  const value = useMemo((): LibraryAuthZContextType => ({
    username: currentUser.username,
    libraryId,
    libraryName: libraryMetadata.title,
    libraryOrg: libraryMetadata.org,
    roles: [],
    permissions: [],
    canManageTeam: currentUser.roles.includes('admin'),
  }), [libraryId, currentUser.username, libraryMetadata?.title]);
```

Note that `libraryMetadata.org` cannot change so likely is safe to omit.